app/Http/Controllers/SettingsController.php

```diff
@@ -896,6 +900,113 @@ public function postLabels(Request $request)
 
         return redirect()->back()->withInput()->withErrors($setting->getErrors());
     }
+    /**
+     * Converts the labels measurement unit for measurable fields in the labels settings
+     *@author [G. Martinez] [<gmartinez@grokability.com>]
+     *
+     *
+     *
+     */
+        public function measurementConverter($setting, Request $request)
+        {
+            \Log::info('This is the old setting:'.$request->old('labels_measurement_type'));
+            switch ($setting->labels_measurement_type) {
+
+                case "in":
+//                      converts centimeters to inches
+                    if ($request->old($setting->labels_measurement_type) == 'cm') {
+
+                        $setting->labels_width = ($setting->labels_width) / 2.54;
```

Comment:
And definitely you should really fix this "magic constant" problem - generally we don't like using "magic numbers" - you should use a variable instead. `$cm_per_inch` = 2.54, and then do your division by `$cm_per_inch`.

Once you make that change I think you're going to see another optimization we can do...